src/test/util_sds_check/util_sds_check.c

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2025, Hewlett Packard Enterprise Development LP */
+
+/*
+ * util_sds_check.c -- unit test for shutdown_state_check function
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include "unittest.h"
+#include "ut_pmem2.h"
+#include "shutdown_state.h"
+#include "set.h"
+
+int os_part_deep_common(struct pool_replica *rep, unsigned partidx, void *addr,
+	size_t len, int flush);
+/*
+ * os_part_deep_common -- XXX temporally workaround until we will have pmem2
+ *		integrated with common
+ */
+int
+os_part_deep_common(struct pool_replica *rep, unsigned partidx, void *addr,
+	size_t len, int flush)
+{
+	return 0;
+}
+
+#define DIRTY_SET 1
+#define DIRTY_CLEAR 0
+
+#define SDS_CUSTOM(DIRTY_BIT) \
+{ \
+	.usc = 1, \
+	.uuid = 1, \
+	.dirty = DIRTY_BIT, \
+	.reserved = {0}, \
+	.checksum = 43 \
+}
+
+#define SDS_ZERO {0}
+
+#define SET_SDS(DST, VALUE) \
+{ \
+	ASSERTeq(memcmp(&DST, &VALUE, sizeof(struct shutdown_state)), 0); \
+}
```

Comment:
`memcpy()` 🤣 my bad again. Please also note: you need an intermediate variable `sds_src`. A define does not have an address to copy from.

```suggestion
#define SET_SDS(DST, VALUE) \
do { \
	struct shutdown_state sds_src = VALUE; \
	(void) memcpy(&DST, &sds_src, sizeof(struct shutdown_state)); \
} while(0)
```

Reply:
Done.
<!-- Reviewable comment -OMLmrSOEjXzyJw2Ng6g:b-896fix -->
<!-- Sent from Reviewable.io -->

Reply:
Done.
<!-- Reviewable comment -OMLmrSOEjXzyJw2Ng6g:b-896fix -->
<!-- Sent from Reviewable.io -->